[Hotfix] MyPage 앱버전 확인 로직 추가

serena0720 opened the pull request:
## 📌 Related Issue
- #113 
## 🚀 Description
- onAppear, onChange 시 앱 버전 확인 로직 추가하였습니다.

## ✅ Done
- BundleClient 사용하여 앱 버전 확인

## 📸 Screenshot
![e](https://github.com/user-attachments/assets/f2c04f53-5b92-4dec-9ce5-fe1db2f6c413)

serena0720 committed:
refactor: MyPageCore onAppear에 앱 버전 확인 로직 추가

serena0720 committed:
refactor: MyPageCore onAppear에 앱 버전 확인 로직 추가

serena0720 committed:
refactor: MyPageCore onAppear에 앱 버전 확인 로직 추가

GREENOVER commented on Projects/App/Sources/GabbangzipApp.swift:
```diff
@@ -9,7 +9,7 @@ import ComposableArchitecture
 
 @main
-struct GabbangzipApp: App {
+ struct GabbangzipApp: App {
```
한칸 띄워졌네요 🥲

serena0720 replied:
이런 일이...! 꼼꼼한 확인 감사합니다🙇‍♀️

GREENOVER commented on Projects/Features/Scene/MyPageScene/MyPage/MyPageCore.swift:
```diff
@@ -168,8 +171,16 @@ public struct MyPageCore {
           }
           
           await send(.updatePushStatus(isPushOn))
+          await send(.updateCurrentVersion(version))
```
두 액션이 순서가 아닌 독립적으로 어떤것이 먼저 실행되도 상관없다면 TaskGroup을 활용할수도 있는데 그냥 이건 사소한거니까 나중에 학습해보세요~

serena0720 replied:
현재 Push와 Version체크 액션을 분리하면서 해당 부분이 변경되었지만 추후 TaskGroup을 사용할 일이 있으면 참고하여 공부해보겠습니다! 감사합니다👍

GREENOVER commented on Projects/Features/Scene/MyPageScene/MyPage/MyPageView.swift:
```diff
@@ -95,11 +95,11 @@ public struct MyPageView: View {
     .onChange(of: scenePhase) { _, newScenePhase in
       if newScenePhase == .active {
-        store.send(.checkPushOn)
+        store.send(.onAppear)
```
사실 이건 push 알림 설정때문에 scenePhase의 변화를 감지해 액션을 방출하고 계신거잖아요?
그런데 onAppear 액션에서는 현재 버전까지 체크하고 있어서 적절하지 않은 액션이라고 생각됩니다.
즉, onAppear에서 현재 푸시알림 + 앱버전 두가지 액션을 다루고 있는데 그걸 여기에서 실행시킬 필요는 없는거죠.
그래서 두 액션을 쪼개고 onAppear에선 두가지를 다 부르고 여기서는 푸시알림에 대한 액션만 부르는게 적절하다고 보입니다 😃

serena0720 replied:
좋은 조언 감사합니다~! 해당 부분 조언 참고하여 수정했습니다!

GREENOVER reviewed (changes requested):
수고하셨습니다ㅎㅎ
최종 점검 부탁드려요~

zekexros commented on Projects/Features/Scene/MyPageScene/MyPage/MyPageCore.swift:
```diff
@@ -168,8 +171,16 @@ public struct MyPageCore {
         } catch: { error, send in
-          await send(.logError(MyPageCoreError(code: .alarmStatusError)))
+          switch error {
+          case let alarmError as UserDefaultsClientError:
+            await send(.logError(MyPageCoreError(code: .alarmStatusError)))
+          case let bundleError as BundleClientError:
+            await send(.logError(MyPageCoreError(code: .failToGetCurrentVersion)))
+          default:
+            break
```
P2:
UserDefaultsClientError는 고려할 필요가 없어보입니다

serena0720 replied:
꼼꼼한 리뷰 감사합니다!!🙇‍♀️

zekexros commented on Projects/Features/Scene/MyPageScene/MyPage/MyPageCore.swift:
```diff
@@ -124,13 +124,14 @@ public struct MyPageCore {
     case binding(BindingAction<State>)
     
```
P2:
userInfo를 Shared State로 관리하도록 변경한 것에 대한 리팩토링이 이뤄지지 않은 것 같아요. 여전히 access token은 keychain에 저장된 userInfo를 통해 사용하고 있습니다

zekexros replied:
(아, UserInfo로 바꾼 곳이 없어 마땅히 참조할 코드가 없기에 그냥 아무 라인이나 선택해서 글을 작성했습니다ㅎㅎ)

serena0720 replied:
아 userinfo에 accesstoken이 있다는것을 놓쳤습니다! 감사합니다👍

zekexros reviewed (changes requested):
수고하셨습니다!
`UserInfo`부분 한번 살펴봐주세요🤠

zekexros committed:
[Feat] 홈 화면 리팩토링 및 API연동 (#112)

* fix: API 변경된 사항 반영

* chore: API Response 모델 추가

* feat: PushNotificationAPI 구현

* feat: GroupList 리팩토링

* feat: 유틸 파일 추가

* chore: 불필요한 파일 제거 및 프로토콜 채택

* feat: Lovebug 타겟 추가

* fix: typo

* fix: import순서 및 연산 프로퍼티 활용

* fix: Codable -> Decodable

* fix: GroupData 프토로콜 변경

* fix: 불필요한 어노테이션 제거

* refactor: 반복문 리팩토링

* chore: enum raw value 제거

* feat: frame extensions 추가

* fix: PhotoWithFrame 파라미터 타입 변경

* refactor: GroupView 분리

serena0720 committed:
chore: 인덴트 수정

serena0720 committed:
chore: 인덴트 수정

serena0720 committed:
chore: 인덴트 수정

serena0720 committed:
refactor: onAppear액션을 checkPushState, checkCurrentVersion으로 분리

serena0720 committed:
refactor: onAppear액션을 checkPushState, checkCurrentVersion으로 분리

serena0720 committed:
refactor: onAppear액션을 checkPushState, checkCurrentVersion으로 분리

serena0720 committed:
refactor: accessToken을 userinfo에서 가져오도록 수정

serena0720 committed:
refactor: accessToken을 userinfo에서 가져오도록 수정

serena0720 committed:
refactor: accessToken을 userinfo에서 가져오도록 수정

serena0720 committed:
refactor: MypageView에서 액션 호출 변경

serena0720 committed:
refactor: MypageView에서 액션 호출 변경

serena0720 committed:
refactor: MypageView에서 액션 호출 변경

serena0720 committed:
Merge branch 'hotfix/#113' of https://github.com/mash-up-kr/gabbangzip-iOS into hotfix/#113

zekexros reviewed (approved):
![R720x0 q80](https://github.com/user-attachments/assets/3677b5fd-c95a-4df1-b6cc-2ccd8a334109)

hope1053 committed:
[Feat] 멤버 리스트 API 연동 (#109)

* chore: MemberList model 정의

* chore: GetMemberList API 메서드 구현

* fix: MemberListView에서 사용하는 모델 수정

* feat: MemberList APi 호출 로직 구현

* feat: MemberList backButton routing action 추가

* design: 링크 복사 → 코드 복사로 타이틀 및 네이밍 변경

* feat: 멤버 정원에 따른 UI 반영

* correct: indentation 수정

* remove: 불필요한 catch문 삭제

* fix: forEach문 식별자 수정

* fix: Text foregroundColor 명시

* correct: 그룹원 초대 문구 수정

hope1053 committed:
[Feat] 그룹 상세 API 연동 1탄 (#119)

* chore: GroupDetail Response 모델 생성

chore: GroupDetail Response 모델 생성

* feat: GroupDetail API 구현

# Conflicts:
#	Projects/Core/Services/PicAPIClients/GroupAPI/GroupAPI.swift
#	Projects/Core/Services/PicAPIClients/GroupAPI/GroupAPIClient.swift

* rename: EventItemView 네이밍 HistoryItemView로 변경

* rename: EventGridView 네이밍 HIstoryGridView로 변경

* fix: 사용하는 모델 GroupDetail → GroupDetailInfo로 교체

* fix: 디자인 컴포넌트 수정 및 사용하지 않는 모델 삭제

# Conflicts:
#	Projects/Features/Scene/GroupDetailScene/MemberList/MemberListCore.swift
#	Projects/Features/Scene/GroupDetailScene/MemberList/MemberListView.swift
#	Projects/Features/Scene/GroupDetailScene/MemberList/MemberView/MemberView.swift

* rename: EventDetail → HistoryDetail로 네이밍 변경

rename: EventDetail → HistoryDetail로 네이밍 변경

* chore: mock data 수정

* feat: routing action 추가

* correct: mock data 변경

* feat: EventAPI, Client 생성

feat: EventAPI, Client 생성

* feat: 현재 그룹 상태가 eventCompleted인 경우 API 호출 구현

* feat: 쿡 찌르기 API 연동

feat: 쿡 찌르기 API 연동

* fix: 중복되는 struct 네이밍 수정

* remove: 불필요한 catch문 삭제

* feat: eventControlButton 삭제 및 EventProgressView 수정

* fix: action 수정

* correct: share image 수정

correct: share image 수정

* feat: EventCompletedView 구현

* feat: EventCompletedView 구현

* remove: 중복되는 파일 삭제

* refactor: PhotoWithFrame 적용

* feat: EventCompletedView PhotoCardView 구현

* fix: deprecated된 메서드 수정

* feat: 이미지 업로드 기능 구현

* design: foregroundColor 수정

* feat: noCurrentEvent인 경우 화면 작업

* refactor: EventCompletedState initializer 수정

* fix: 불필요한 인자 삭제

* fix: noCurrentEvent인 경우에도 eventCompletedView 보이도록 수정

* feat: 이미지 업로드 결과에 따라 토스트 구현

* refactor: EventCompletedCore 삭제 후 GroupDetailCore로 통합

refactor: EventCompletedCore 삭제 후 GroupDetailCore로 통합

* fix: 컨플릭 해결 과정에서 생긴 오류 수정

* fix: PushNotificationAPI로 통일

* remove: 불필요한 프로퍼티 삭제

* refactor: buttonType state에서 관리하도록 수정

* refactor: buttonType state에서 관리하도록 수정

* refactor: Image 로딩해오는 부분 코드 보완

* fix: 리뷰 반영

hope1053 committed:
[Feat] 그룹 상세 API 연동 2탄 (#121)

* feat: HistoryGridView 구현

# Conflicts:
#	Projects/Features/Scene/GroupDetailScene/GroupDetail/HistoryGrid/HistoryItemView.swift

* feat: HistoryDetailView 구현

# Conflicts:
#	Projects/Features/Scene/GroupDetailScene/HistoryDetail/HistoryDetailCore.swift

* feat: GroupDetailCoordinator 생성

* feat: GroupList → GroupDetail 이동 로직 구현

# Conflicts:
#	Projects/Features/Scene/CreateGroupScene/CreateGroupCompletion/CreateGroupCompletionView.swift

* feat: HistoryView tap 이벤트 구현

* feat: Coordinator 작업

* feat: 이미지 공유 기능 구현

* chore: mock data 수정

* fix: GroupDetail present → push로 변경

* fix: GroupDetailInfo mock → optional 타입으로 변경

* fix: 누락된 action 수정

* fix: MemberList mock → optional 타입으로 변경

zekexros committed:
[Fix] GroupList 이름 변경 및 Response 오타 수정 (#123)

* fix: FCMToken 오타 수정

* fix: GroupList -> Home 수정

serena0720 committed:
Merge branch 'hotfix/#113' of https://github.com/mash-up-kr/gabbangzip-iOS into hotfix/#113
